### ISSUE TITLE
Orika is not compatible with hibernate 4.1.9.Final

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#73]
```
What steps will reproduce the problem?
1. Start test sush as attached (Issue 20)


What is the expected output? What do you see instead?
I see internal exception in mapper

What version of the product are you using? On what operating system?
Orika 1.3.5 and 1.4.0

Please provide any additional information below.
hibernate 4.1.9.Final has dependency on javassist-3.17.0-GA
Оrika  is not compatible with javassist-3.17.0-GA.

```

Original issue reported on code.google.com by `dkhomya...@gmail.com` on 19 Dec 2012 at 10:20

**Comment by GoogleCodeExporter** [comment]
```
I've made mistake, problem with javassist-3.17.1
```

Original comment by `dkhomya...@gmail.com` on 19 Dec 2012 at 12:20

**Comment by GoogleCodeExporter** [comment]
```
Are you sure that this is not a bug with the javassist 3.17.1 version?
It appears that Orika is hit by this issue: 
https://issues.jboss.org/browse/JASSIST-186 
```

Original comment by `matt.deb...@gmail.com` on 29 Dec 2012 at 8:12

**Comment by GoogleCodeExporter** [comment]
```
We'll leave it open as a reminder...
```

Original comment by `matt.deb...@gmail.com` on 30 Dec 2012 at 6:16
- Added labels: **Type-Other**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
It looks like this issue is corrected by upgrading to javassist 3.18.0-GA, and 
in addition, the referenced issue on the Javassist JIRA was marked resolved on 
3.18.0-GA, so marking this as 'Done'.
```

Original comment by `matt.deb...@gmail.com` on 7 Jul 2013 at 3:58
- Changed state: **Done**
